### ISSUE TITLE
Navigation icon back arrow goes back to hamburger when NavigationPage is set to MasterDetailPage.Detail.

### ISSUE THREAD

**Issue by mikepham** [nativecode-dev/oss-xamarin#5]
**Expectation**:
1) Open MasterDetailPage menu. Hamburger icon becomes back arrow.
2) Push a new page onto the navigation stack and the back arrow should stay, indicating to the user they can navigate backwards.

The following meets the expectation:

``` csharp
new NavigationPage(new MasterDetailPage() { Detail = detail, Master = master });
```

The following does not:

``` csharp
new MasterDetailPage() { Detail = new NavigationPage(detail), Master = master };
```

The back arrow changes back to a hamburger menu and the user is left thinking the navigation drawer should open again, but instead the page action is to pop the page off the stack.

**Comment by dhaligas** [comment]
I see what you referring to.  I usually set a new Nav Detail Page when selecting an item from the menu view

**Comment by mikepham** [comment]
I'm not quite sure which method is correct as I've seen it done both ways. Personally, I prefer wrapping the MasterDetailPage in a NavigationPage, but some people wrap the detail page. There doesn't seem to be a "correct" answer, so it makes it hard to fix.

**Comment by mikepham** [comment]
Fixed in [6f2917d](https://github.com/nativecode-dev/oss-xamarin/commit/6f2917d0c1d30bd4635c3189e78ba77f5d80d083).
